src/scripting/effects.cpp

```diff
@@ -5477,14 +5499,26 @@ uint32_t internal_execute_effect(EFFECT_PARAMTERS) {
 
 void execute(sys::state& state, dcon::effect_key key, int32_t primary, int32_t this_slot, int32_t from_slot, uint32_t r_lo,
 		uint32_t r_hi) {
-	bool els = false;
+	effect_context els;
 	internal_execute_effect(state.effect_data.data() + state.effect_data_indices[key.index() + 1], state, primary, this_slot, from_slot, r_lo, r_hi, els);
+	for(const auto& p : els.future_n_event) {
+		event::trigger_national_event(state, p.e, p.n, p.r_lo, p.r_hi, p.from_slot, p.ft);
+	}
+	for(const auto& p : els.future_p_event) {
+		event::trigger_provincial_event(state, p.e, p.p, p.r_lo, p.r_hi, p.from_slot, p.ft);
+	}
 }
 
 void execute(sys::state& state, uint16_t const* data, int32_t primary, int32_t this_slot, int32_t from_slot, uint32_t r_lo,
 		uint32_t r_hi) {
-	bool els = false;
+	effect_context els;
 	internal_execute_effect(data, state, primary, this_slot, from_slot, r_lo, r_hi, els);
+	for(const auto& p : els.future_n_event) {
+		event::trigger_national_event(state, p.e, p.n, p.r_lo, p.r_hi, p.from_slot, p.ft);
+	}
+	for(const auto& p : els.future_p_event) {
+		event::trigger_provincial_event(state, p.e, p.p, p.r_lo, p.r_hi, p.from_slot, p.ft);
```

Comment:
so, this won't work. Suppose that one of the stored provincial events fires a national event (I have seem mods do this). That event will now never fire since it is added to the future_n_event list which will no longer be checked at this point.

Reply:
I also don't like how we are creating new vectors (and hence new allocations) constantly. We probably can't just make the variable static because of the possibility of recursion, but I wish something could be done here.

Reply:
uhm, excuse me? it does work

Reply:
check the logic above for pushing events to the queue

Reply:
I did check the logic

Reply:
Province event in future_p_event adds an event to the future_n_event vector. That vector is never checked again. Hence the event is never fired

Reply:
i dont get what you mean then, the events do fire because their everything is pushed to the local effect context

Reply:
oh, so each gets its own local effect context ... then this doesn't fix the recursion problem. event fires -> local context -> fires event -> sub local context -> etc

Reply:
> Province event in future_p_event adds an event to the future_n_event vector. That vector is never checked again. Hence the event is never fired

Ok that is wrong
a) the effect within the event will have ITS OWN effect context
b) so this will occur:

event 100 gets fired (province)
ok good, now we execute the effect of event 100
we execute whatever, immediate or option 0 etc, MAKING OUR OWN LOCAL EFFECT CONTEXT
this event fires event 500 (national)
this EFFECT LOCAL CONTEXT is independent from the topmost one, it will execute event 500

Reply:
effect_context is LOCAL to each execution unit - basically i made it so events can recursively call other events, but with a specific delay

Reply:
Yeah, I see what you have done now. But the problem is that it doesn't fix the recursion isue